src/metrics.rs

```diff
@@ -55,66 +59,103 @@ pub(crate) const BUCKET_FACTOR: f64 = 2.0;
 /// care about granularity past 1 second.
 pub(crate) const BUCKET_COUNT: usize = 13;
 
-pub(crate) static PROCESSING_TIME: Lazy<HistogramVec> = Lazy::new(|| {
-    prometheus::register_histogram_vec_with_registry! {
-        prometheus::histogram_opts! {
-            "packets_processing_duration_seconds",
-            "Total processing time for a packet",
-            prometheus::exponential_buckets(BUCKET_START, BUCKET_FACTOR, BUCKET_COUNT).unwrap(),
-        },
-        &[DIRECTION_LABEL],
-        registry(),
-    }
-    .unwrap()
-});
-
-pub(crate) static BYTES_TOTAL: Lazy<IntCounterVec> = Lazy::new(|| {
-    prometheus::register_int_counter_vec_with_registry! {
-        prometheus::opts! {
-            "bytes_total",
-            "total number of bytes",
-        },
-        &[DIRECTION_LABEL],
-        registry(),
-    }
-    .unwrap()
-});
-
-pub(crate) static ERRORS_TOTAL: Lazy<IntCounterVec> = Lazy::new(|| {
-    prometheus::register_int_counter_vec_with_registry! {
-        prometheus::opts! {
-            "errors_total",
-            "total number of errors sending packets",
-        },
-        &[DIRECTION_LABEL],
-        registry(),
-    }
-    .unwrap()
-});
-
-pub(crate) static PACKETS_TOTAL: Lazy<IntCounterVec> = Lazy::new(|| {
-    prometheus::register_int_counter_vec_with_registry! {
-        prometheus::opts! {
-            "packets_total",
-            "Total number of packets",
-        },
-        &[DIRECTION_LABEL],
-        registry(),
-    }
-    .unwrap()
-});
-
-pub(crate) static PACKETS_DROPPED: Lazy<IntCounterVec> = Lazy::new(|| {
-    prometheus::register_int_counter_vec_with_registry! {
-        prometheus::opts! {
-            "packets_dropped",
-            "Total number of dropped packets",
-        },
-        &[DIRECTION_LABEL, "reason"],
-        registry(),
+#[derive(Clone, Copy, Debug)]
+pub enum Direction {
+    Read,
+    Write,
+}
+
+impl Direction {
+    pub(crate) const LABEL: &'static str = DIRECTION_LABEL;
+
+    pub fn label(self) -> &'static str {
+        match self {
+            Self::Read => READ_DIRECTION_LABEL,
+            Self::Write => WRITE_DIRECTION_LABEL,
+        }
     }
-    .unwrap()
-});
+}
+
+pub fn processing_time(direction: Direction) -> Histogram {
+    static PROCESSING_TIME: Lazy<HistogramVec> = Lazy::new(|| {
+        prometheus::register_histogram_vec_with_registry! {
+            prometheus::histogram_opts! {
+                "packets_processing_duration_seconds",
+                "Total processing time for a packet",
+                prometheus::exponential_buckets(BUCKET_START, BUCKET_FACTOR, BUCKET_COUNT).unwrap(),
+            },
+            &[Direction::LABEL],
+            registry(),
+        }
+        .unwrap()
+    });
+
+    PROCESSING_TIME.with_label_values(&[direction.label()])
+}
+
+pub fn bytes_total(direction: Direction) -> IntCounter {
+    pub(crate) static BYTES_TOTAL: Lazy<IntCounterVec> = Lazy::new(|| {
```

Comment:
Curious why these are `pub(crate)` - is that just leftover from the previous declarations?